updated readme (#41)
alcpereira/100hunters: README.md

# 100Hunters

> Group project

Reviewer: ```suggestion
```

# Features
- **Customizable Columns:** The Kanban board will consist of customizable columns representing different stages of the job search process, such as "Applied", "Interview Scheduled", "Offer Received", etc. Users can add, remove, or modify columns based on their needs.

- **Task Management:** Users can create tasks/cards within each column to represent job applications, networking opportunities, follow-ups, or any other relevant activities. These cards can include details such as company name, job title, application status, contact information, and deadlines.

- **Drag-and-Drop Functionality:** The board will support drag-and-drop functionality, allowing users to easily move cards between columns to reflect changes in the status of their job search activities. For example, moving a card from "Applied" to "Interview Scheduled" when a job interview is confirmed.

- **Deadline Reminders:** The board will provide the option to set deadlines for tasks, with optional reminders to ensure users stay on track and meet important application deadlines.

# Technologies used

> - **Frontend:** Tailwind CSS, Typescript (React.js  
> - **Backend:** Remix (SSR React) + Vite + Express, Turso (for database),Storybook, Vitest + React Testing Library + Playwright

Reviewer: ```suggestion
- **Language:** TypeScript
- **Framework:** Remix (React.js) with Vite and Express
- **Frontend libraries:** Tailwind CSS, Storybook
- **Database:** Turso with libSQL (in memory, and hosted as a service)
- **Testing:**  Vitest, React Testing Library, and Playwright
```

# Folder structure
├── app <br>
│   ├── components<br>
│   │   ├── Card<br>
│   │   └── Column<br>
│   ├── db<br>
│   └── routes<br>
│       └── board<br>
├── build<br>
│   ├── client<br>
│   │   └── assets<br>
│   └── server<br>
│       └── assets<br>
├── coverage<br>
├── docs<br>
├── public<br>
└── utils<br>
    ├── mermaid<br>

>     └── seed<br>

Reviewer: You can use the triple backtick (\`\`\`) around this block to have a nice formatting (and remove the `<br>`)

I'd also remove: 
- `Card` and `Column`, they are implementation details
- `client` and `server` folders with their `assets`, also an implementation detail

No need for more details for now, as we have an Issue to add more details to this structure (see #22)

# Setup Instructions
1. Clone the repository to your local machine.
2. Navigate to the project directory.
3. Install dependencies using npm install.
4. Set up the backend server and database.
5. Run the frontend application using npm start.
6. Access the application through your web browser.
7. Sign up or log in to start using the Kanban board for managing your job hunt.

# Contribute
Contributions to the project are welcome! If you'd like to contribute, please follow these guidelines:

> (Link to contribute .env)

Reviewer: ```suggestion
[Contributing guide](CONTRIBUTING.md)
```

# License
This project is licensed under the MIT License

# To go further
- Login (Discord OAuth)
- Add/rename columns
- List of companies (Reviews)

> - Analytics

Reviewer: I think we can delete those, they will be present in the `docs/features.md` file (see #13)
